docs: add 0.9.0 Changelog (#5308)
confluentinc/ksql: CHANGELOG.md

# Change Log

> ## [0.9.0](https://github.com/confluentinc/ksql/releases/tag/v0.9.0-ksqldb) (2020-05-08)

Reviewer: nit: don't forget to update the date here to the actual release date, if that turns out to not be today.

Author: done

### Features

* add multi-join expression support ([#5081](https://github.com/confluentinc/ksql/pull/5081)) ([002cd5a](https://github.com/confluentinc/ksql/commit/002cd5ad6473cbb000291e32ebe5a459d2870b61))
* support more advanced suite of LIKE expressions ([#5013](https://github.com/confluentinc/ksql/pull/5013)) ([67cd9d9](https://github.com/confluentinc/ksql/commit/67cd9d91690039cc01f43438db1fbcd3fe4924ff))
* add COALESCE function ([#4829](https://github.com/confluentinc/ksql/pull/4829)) ([251c237](https://github.com/confluentinc/ksql/commit/251c237c455f4cf3843628f41e4bbf57795cfd12))

> * add internal request to KsqlRequestConfig and SessionProperties ([#4771](https://github.com/confluentinc/ksql/pull/4771)) ([fc10cae](https://github.com/confluentinc/ksql/commit/fc10caeb6c164febd8041bc523e765fa6a6b77d8))

Reviewer 2: This can be removed this from the features list.

Author: Understand now. 
I removed the rest of the features.

Reviewer: I don't understand. Why did we remove a bunch of features from the list: ksqlquerystatus, muti-way joins, pull query rate limiting, klip 14, scatter gather, and others?

Author: I put them back. I misunderstood the `this` with `these`, so removed everything below. I also removed some the lines related to vert.x changes as they're not new features.

* add KsqlQueryStatus to decouple from KafkaStreams.State ([#5029](https://github.com/confluentinc/ksql/pull/5029)) ([e8cbcde](https://github.com/confluentinc/ksql/commit/e8cbcde548e1f3078b1396173a3f55f86ee20626))
* support multi-way joins ([#5064](https://github.com/confluentinc/ksql/pull/5064)) ([64dd39e](https://github.com/confluentinc/ksql/commit/64dd39ee77928221f08371d5e4424839d20c09f6))
* Adds rate limiting to pull queries ([#4951](https://github.com/confluentinc/ksql/pull/4951)) ([6284111](https://github.com/confluentinc/ksql/commit/6284111652252b63cfc233bdfc3e08dbff983bbd))
* CORS support for Vert.x based ksqlDB ([#4944](https://github.com/confluentinc/ksql/pull/4944)) ([5632893](https://github.com/confluentinc/ksql/commit/56328931d7eb26699fb5bd0940cfd03f7f9e53f9))

> * Do not allow access to new streaming endpoints using HTTP1.x ([#5193](https://github.com/confluentinc/ksql/pull/5193)) ([8b90035](https://github.com/confluentinc/ksql/commit/8b90035c6facf47d81dfd1616784b191141dce31))

Reviewer: This should either be moved to the bug fix section or dropped entirely.

Author: Done

* fail startup if command contains incompatible version ([#5104](https://github.com/confluentinc/ksql/pull/5104)) ([a1751b1](https://github.com/confluentinc/ksql/commit/a1751b1689532ef9abd436d5f27fe9aa8ff555ac))
* Implement cluster health check endpoints ([#5126](https://github.com/confluentinc/ksql/pull/5126)) ([fbc9bf0](https://github.com/confluentinc/ksql/commit/fbc9bf019ec53b02af11832e12ed944ae4e424d0))
* klip-14 - rowtime as pseduocolumn ([#5150](https://github.com/confluentinc/ksql/pull/5150)) ([d541420](https://github.com/confluentinc/ksql/commit/d541420acf61f0e0e7b25d91120886007dbdae01))

> * Migrate /ksql and /ksql/terminate endpoints ([#4851](https://github.com/confluentinc/ksql/pull/4851)) ([af6c792](https://github.com/confluentinc/ksql/commit/af6c7922d3c1b0afdd87e9a079e6f95be9780172))

Reviewer: This should be removed.

Author: Done

* scatter gather query status from all servers in cluster for 'SHOW QUERIES [EXTENDED]' statement ([#4875](https://github.com/confluentinc/ksql/pull/4875)) ([7385a31](https://github.com/confluentinc/ksql/commit/7385a31dd33293b76e594649213da2abb81b8ddf))

> * speed up restarts by not building topologies for terminated queries ([#5002](https://github.com/confluentinc/ksql/pull/5002)) ([2472382](https://github.com/confluentinc/ksql/commit/24723824635e7ca175bc5be1fbf6a415010b00bb))

Reviewer: This is a bug fix, not a feature.

Author: done

* Support wildcards for unauthed paths ([#4945](https://github.com/confluentinc/ksql/pull/4945)) ([9173f1d](https://github.com/confluentinc/ksql/commit/9173f1dbf92fea0f1b17b6c85fb8a6c25e789ce3))
* transient queries added to show queries output ([#5105](https://github.com/confluentinc/ksql/pull/5105)) ([e8a2a63](https://github.com/confluentinc/ksql/commit/e8a2a63210219cbac7d979ec0915f1d2016398e2))

### Bug Fixes

> * 'drop (stream|table) if exists' fails if source does not exist ([#4872](https://github.com/confluentinc/ksql/pull/4872)) ([b0669a0](https://github.com/confluentinc/ksql/commit/b0669a044b53407fba3ceb11bddee73e69c0006f))

Reviewer: Can we reword this to "'drop (stream|table) if exists' should not fail if source does not exist"? Otherwise it makes it sound like the change was to make it start failing, which is the opposite of what the fix actually does.

Author: done

> * add content type header to ported endpoints response ([#5054](https://github.com/confluentinc/ksql/pull/5054)) ([1ea80d4](https://github.com/confluentinc/ksql/commit/1ea80d4956c6588ec530a90ab822b2328b524ce7))

Reviewer: This can be dropped.

Author: done

* CommandRunner metric has correct metric displayed when thread dies ([#4653](https://github.com/confluentinc/ksql/pull/4653)) ([1db542b](https://github.com/confluentinc/ksql/commit/1db542bfea5006fc9b3af59bca28d4af48fc5d57))
* do not allow GROUP BY and PARTITION BY on boolean expressions ([#4940](https://github.com/confluentinc/ksql/pull/4940)) ([d84d2d1](https://github.com/confluentinc/ksql/commit/d84d2d110d5ac502de4ba16602aa32babe24b4ca))
* do not allow grouping sets ([#4942](https://github.com/confluentinc/ksql/pull/4942)) ([51bb9f6](https://github.com/confluentinc/ksql/commit/51bb9f6f4bcef2d75f6d6d96c4c883559a21999b))
* do not allow implicit casting in UDAF function lookup ([#5145](https://github.com/confluentinc/ksql/pull/5145)) ([6709010](https://github.com/confluentinc/ksql/commit/6709010b06fed487334338e5e102bf53d51b4c46))
* do not filter out rows where PARTITION BY resolves to null ([#4823](https://github.com/confluentinc/ksql/pull/4823)) ([e75a792](https://github.com/confluentinc/ksql/commit/e75a79286927a6e43b7d580b32215fb18b00d707))

> * Don't set chunked transfer encoding for HTTP2 proxy ([#4985](https://github.com/confluentinc/ksql/pull/4985)) ([3f16844](https://github.com/confluentinc/ksql/commit/3f1684400dfcdc8d1daafe8feb9363efed9327d8))

Reviewer: This can be dropped.

Author: done

> * Don't wait for streams thread to be in running state ([#4908](https://github.com/confluentinc/ksql/pull/4908)) ([ceeead2](https://github.com/confluentinc/ksql/commit/ceeead21d32232a2e30d79b84342a0dec3c36fd4))

Reviewer: This can be dropped. It's present in 0.8.1.

Author: done

* If startup hangs (esp on preconditions), shutdown server correctly ([#4889](https://github.com/confluentinc/ksql/pull/4889)) ([20c4b59](https://github.com/confluentinc/ksql/commit/20c4b5903c13fb8435aa94e486b637074b5bb769))
* Improve error message for where/having type errors ([#5023](https://github.com/confluentinc/ksql/pull/5023)) ([23eb80d](https://github.com/confluentinc/ksql/commit/23eb80d06ea0275b28edac9b0fb5453ea4a966f8))
* improve handling of NULLs ([#5019](https://github.com/confluentinc/ksql/pull/5019)) ([c53dd68](https://github.com/confluentinc/ksql/commit/c53dd68bd912d79ced54403650ba13597ffe35a0)), closes [#4912](https://github.com/confluentinc/ksql/issues/4912)
* include lower-case identifiers among those that need quotes ([#3723](https://github.com/confluentinc/ksql/pull/3723)) ([#5139](https://github.com/confluentinc/ksql/issues/5139)) ([3bcbcf4](https://github.com/confluentinc/ksql/commit/3bcbcf45e1fc03245a57005d318133c568fb12c6))

> * Infer TLS based on scheme of server string ([#4893](https://github.com/confluentinc/ksql/pull/4893)) ([b80a14c](https://github.com/confluentinc/ksql/commit/b80a14c1ddcbe9d90e2e72901dc76a96fa26ac20))

Reviewer: This can be dropped. It's present in 0.8.1.

Author: done

> * json creator for QueryStatusCount should match getter (MINOR) ([#5052](https://github.com/confluentinc/ksql/pull/5052)) ([988c3d1](https://github.com/confluentinc/ksql/commit/988c3d1068b4cf970ea6c31fecf58496b7c7ee72))

Reviewer: This can be dropped. It fixes a new feature introduced in this release.

Author: done

> * make CLI unit-tests platform independent ([#4743](https://github.com/confluentinc/ksql/pull/4743)) ([3dc664f](https://github.com/confluentinc/ksql/commit/3dc664fe4d76ce917933c64a4673404e660346ff))

Reviewer: This can be dropped. Seems to be about tests only.

Author: done

* make endpoints available while waiting for precondition ([#5069](https://github.com/confluentinc/ksql/pull/5069)) ([1136162](https://github.com/confluentinc/ksql/commit/113616260cc0db89124330ed5e935e4b9e5af281))
* Make sure internal client is configured for TLS ([#5059](https://github.com/confluentinc/ksql/pull/5059)) ([37c7713](https://github.com/confluentinc/ksql/commit/37c7713920c7654c9b5c2f81ed72225233e1176c))

> * Move Cors handler in front of /chc handlers ([#5239](https://github.com/confluentinc/ksql/pull/5239)) ([ba03884](https://github.com/confluentinc/ksql/commit/ba03884166c6a57b2258d2e8111a2a5ca1a41917))

Reviewer: This can be dropped. Fixes a feature introduced in this release.

Author: done

> * NPE in latest_by_offset if first element being processed has null… ([#4975](https://github.com/confluentinc/ksql/pull/4975)) ([4c72f93](https://github.com/confluentinc/ksql/commit/4c72f93813af19f6fd074b3a691018434731d545))

Reviewer: ```suggestion
* NPE in latest_by_offset if first two elements processed are both null ([#4975](https://github.com/confluentinc/ksql/pull/4975)) ([4c72f93](https://github.com/confluentinc/ksql/commit/4c72f93813af19f6fd074b3a691018434731d545))
```

Author: done

* only create processing log stream if it doesn't exist ([#4805](https://github.com/confluentinc/ksql/pull/4805)) ([8dead0f](https://github.com/confluentinc/ksql/commit/8dead0f4ea82de1efa34614a715b267f0246f224))
* output valid multiline queries when running SHOW QUERIES ([#4956](https://github.com/confluentinc/ksql/pull/4956)) ([ec74a33](https://github.com/confluentinc/ksql/commit/ec74a337c93e834278938f9fb692aab143d06276))
* query id for TERMINATE should be case insensitive ([#5005](https://github.com/confluentinc/ksql/pull/5005)) ([588c1e9](https://github.com/confluentinc/ksql/commit/588c1e905fc92c40635ca3271089dfecdfd4f3f6))
* reject requests to new API server if server is not ready ([#5048](https://github.com/confluentinc/ksql/pull/5048)) ([d988722](https://github.com/confluentinc/ksql/commit/d9887220bb5a688b6e8682227189341e88a61967))

> * Removes unnecessary error logging for heartbeat since this is ex… ([#4809](https://github.com/confluentinc/ksql/pull/4809)) ([fa84576](https://github.com/confluentinc/ksql/commit/fa845763d8c52c54a65a7618a044300e8f0043f6))

Reviewer: ```suggestion
* Remove unnecessary error logging for heartbeat ([#4809](https://github.com/confluentinc/ksql/pull/4809)) ([fa84576](https://github.com/confluentinc/ksql/commit/fa845763d8c52c54a65a7618a044300e8f0043f6))
```

Author: done

* support quotes in explain statements (MINOR) ([#5142](https://github.com/confluentinc/ksql/pull/5142)) ([bee2fe0](https://github.com/confluentinc/ksql/commit/bee2fe0f523ee7e763e223bd2d1f4fc6f178ffec))
* throw error when column does not exist during INSERT INTO ([#4926](https://github.com/confluentinc/ksql/pull/4926)) ([89e261d](https://github.com/confluentinc/ksql/commit/89e261da5aff6b62901bc7d73c12baecee7bf95c))
* remove immutable properties from headless mode ([#4936](https://github.com/confluentinc/ksql/pull/4936)) ([5550880](https://github.com/confluentinc/ksql/commit/5550880de097255adbe536907d0656ba8deea0cc))
* rename ksqldb in normal logging path (MINOR) ([#4915](https://github.com/confluentinc/ksql/pull/4915)) ([16172ba](https://github.com/confluentinc/ksql/commit/16172ba8c820bb50c838ea820873b3593e062c6c))
* support trailing slashes in listener URLs ([#5076](https://github.com/confluentinc/ksql/pull/5076)) ([e9e0431](https://github.com/confluentinc/ksql/commit/e9e0431f6eafd7ef18dd2cb9e17730ec3ba1e226))

> * backport fixes from query close ([#4819](https://github.com/confluentinc/ksql/pull/4819)) ([76f960b](https://github.com/confluentinc/ksql/commit/76f960bc5702d3995d30ba538956813728160862)), closes [#4643](https://github.com/confluentinc/ksql/issues/4643) [#4658](https://github.com/confluentinc/ksql/issues/4658)

Reviewer: Remove this. It's in 0.8.0

Author: done

* logic when closing ksqlEngine fixed ([#4917](https://github.com/confluentinc/ksql/pull/4917)) ([a217eb9](https://github.com/confluentinc/ksql/commit/a217eb9a8a80b38c970021faad76a4b9fcade609))

> * don't cleanup topics on engine close ([#4658](https://github.com/confluentinc/ksql/pull/4658)) ([#4857](https://github.com/confluentinc/ksql/issues/4857)) ([21e4751](https://github.com/confluentinc/ksql/commit/21e47519ffd8df8cfee083e559ac28f736f058c1))

Reviewer: Remove this. It's in 0.7.1

Author: done

* use describeTopics() for Kafka healtcheck probe ([#4814](https://github.com/confluentinc/ksql/pull/4814)) ([578d0d5](https://github.com/confluentinc/ksql/commit/578d0d543bce685ca6e253a8c6f9bcfbf191093d))

### chore

> * add GROUP BY support for any key names ([#4899](https://github.com/confluentinc/ksql/pull/4899)) ([e7cbdfc](https://github.com/confluentinc/ksql/commit/e7cbdfcc8c9853e2ae6dfcaf670e04f07ccd5444)), closes [#4898](https://github.com/confluentinc/ksql/issues/4898)

Reviewer 2: I think this should go in the features section.

Author: done

> * partition-by primitive key support (#4098) ([7addf88](https://github.com/confluentinc/ksql/commit/7addf8856a6d62a6890a5f2520eead26538233a6)), closes [#4098](https://github.com/confluentinc/ksql/issues/4098)

Reviewer 2: This should also go in the features section.

Author: done

### BREAKING CHANGES

* Select star, i.e. `select *`, no longer expands to include `ROWTIME` column(s). Instead, `ROWTIME` is only included in the results of queries if explicitly included in the projection, e.g. `select rowtime, *`.
This only affects new statements. Any view previously created via a `CREATE STREAM AS SELECT` or `CREATE TABLE AS SELECT` statement is unaffected.

> * This commit changes the system generated column name for any columns in projections that are struct field dereferences. Previously, the full path was used when generating the name, now only the final field name is used. For example, `SELECT someStruct->someField, ...` would previously of generated a column name of `SOMESTRUCT__SOMEFIELD`, and will now generate a name of `SOMEFIELD`. Generated column names may have a numeric appended to the end to ensure uniqueness, for example `SOMEFIELD_2`.

Reviewer 2: "This commit" can be changed to "This release".

Author: done

Note: it is recommended that you do not rely on system generated column names for production systems as there naming may change between releases. Providing an explict alias ensures consistent naming across releases. For example, `SELECT someStruct->someField AS someField`.
Backwards compatibility: existing running queries will not be affected by this change: they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`

> * Existing queries that reference a single GROUP BY column in the projection would fail if they were resubmitted, due to a duplicate column. The same existing queries will continue to run if already running, i.e. this is only a change for newly submitted queries. Existing queries will use the old query semantics.

Reviewer 2: Do this and the subsequent 3 bullets belong underneath the "This commit changes the..." bullet? If so they should be indented underneath it.

Author: This belongs to one commit. But the subsequent 3 bullets belong to another commit. I'll merged the 3 bullets below into one.

Reviewer: This indentation still looks incorrect.

Author: I am not sure what indentation. Anyway, I fixed the one from the `Note: it is reco...` line. I hope that's the one I should've fixed.

Reviewer: I see. These two bullets ("Existing queries that reference ..." and "Any existing persistent queries ... ") are confusing because they don't make sense by themselves, which is why Derek and I thought they were meant to be part of a previous entry, but it turns out they actually aren't -- they were just written in a way that refers to a previous part of the commit message which isn't included in the breaking change section, e.g., https://github.com/confluentinc/ksql/commit/2002458f6b668b7b17b05b192a7cc3fab0a23b20 (cc @big-andy-coates for future reference).

I think we should reword the second one to "Push queries may see changes in column names. Pull queries and any existing persistent queries are unaffected." to help clarify some of the confusion.

> * * Any existing persistent queries, e.g. those created with `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT` or `INSERT INTO`, will be unaffected: their column names will not change.

Reviewer 2: There's an extra bullet character here.

Author: done

# Change Log

## [0.9.0](https://github.com/confluentinc/ksql/releases/tag/v0.9.0-ksqldb) (2020-05-07)

> ### Features

Reviewer 3: Would be worth separating out the truly user visible features like multi-way joins and the improvements on the `LIKE` clause.

Author: Done. 

I put them at the top of the list.

* This commit changes the system generated column name for any columns in projections that are struct field dereferences. Previously, the full path was used when generating the name, now only the final field name is used. For example, `SELECT someStruct->someField, ...` would previously of generated a column name of `SOMESTRUCT__SOMEFIELD`, and will now generate a name of `SOMEFIELD`. Generated column names may have a numeric appended to the end to ensure uniqueness, for example `SOMEFIELD_2`.
Note: it is recommended that you do not rely on system generated column names for production systems as there naming may change between releases. Providing an explict alias ensures consistent naming across releases. For example, `SELECT someStruct->someField AS someField`.
Backwards compatibility: existing running queries will not be affected by this change: they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`

> Co-authored-by: Andy Coates <big-andy-coates@users.noreply.github.com>

Reviewer 3: Why are names and email addresses in this changelog? Doesn't look right?

Reviewer: The changelog generator blindly copy-pastes whatever is in the commit message. This line was part of the commit message, and was picked up as a result: https://github.com/confluentinc/ksql/commit/2002458f6b668b7b17b05b192a7cc3fab0a23b20

Author: I just removed them.

* Existing queries that reference a single GROUP BY column in the projection would fail if they were resubmitted, due to a duplicate column. The same existing queries will continue to run if already running, i.e. this is only a change for newly submitted queries. Existing queries will use the old query semantics.
Co-authored-by: Big Andy Coates <andy@confluent.io>
* * Any existing persistent queries, e.g. those created with `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT` or `INSERT INTO`, will be unaffected: their column names will not change.
* Pull queries will be unaffected.
* Push queries, which rely on auto-generated column names, may see a change in column names.

> 

Reviewer: Let's add a breaking change for the removal of Jetty. Here's some proposed text:

> The ksqlDB server no longer ships with Jetty, which means Jetty-specific dependencies (e.g., certain login modules used for basic authentication) must be supplied via the `KSQL_CLASSPATH` environment variable when starting the server in order to be found.

cc @purplefox to adjust the wording.

Author: Done

* use describeTopics() for Kafka healtcheck probe ([#4814](https://github.com/confluentinc/ksql/pull/4814)) ([578d0d5](https://github.com/confluentinc/ksql/commit/578d0d543bce685ca6e253a8c6f9bcfbf191093d))
* Do not allow access to new streaming endpoints using HTTP1.x ([#5193](https://github.com/confluentinc/ksql/pull/5193)) ([8b90035](https://github.com/confluentinc/ksql/commit/8b90035c6facf47d81dfd1616784b191141dce31))
* speed up restarts by not building topologies for terminated queries ([#5002](https://github.com/confluentinc/ksql/pull/5002)) ([2472382](https://github.com/confluentinc/ksql/commit/24723824635e7ca175bc5be1fbf6a415010b00bb))

### BREAKING CHANGES

* Select star, i.e. `select *`, no longer expands to include `ROWTIME` column(s). Instead, `ROWTIME` is only included in the results of queries if explicitly included in the projection, e.g. `select rowtime, *`.
This only affects new statements. Any view previously created via a `CREATE STREAM AS SELECT` or `CREATE TABLE AS SELECT` statement is unaffected.

> * This release changes the system generated column name for any columns in projections that are struct field dereferences. Previously, the full path was used when generating the name, now only the final field name is used. For example, `SELECT someStruct->someField, ...` would previously of generated a column name of `SOMESTRUCT__SOMEFIELD`, and will now generate a name of `SOMEFIELD`. Generated column names may have a numeric appended to the end to ensure uniqueness, for example `SOMEFIELD_2`.

Reviewer 4: ```suggestion
* This release changes the system generated column name for any columns in projections that are struct field dereferences. Previously, the full path was used when generating the name, now only the final field name is used. For example, `SELECT someStruct->someField, ...` previously generated a column name of `SOMESTRUCT__SOMEFIELD` and now generates a name of `SOMEFIELD`. Generated column names may have a numeral appended to the end to ensure uniqueness, for example `SOMEFIELD_2`.
```

>   Note: it is recommended that you do not rely on system generated column names for production systems as there naming may change between releases. Providing an explict alias ensures consistent naming across releases. For example, `SELECT someStruct->someField AS someField`.

Reviewer 4: ```suggestion
  Note: it is recommended that you do not rely on system generated column names for production systems, because naming logic may change between releases. Providing an explicit alias ensures consistent naming across releases, for example, `SELECT someStruct->someField AS someField`.
```

>   Backwards compatibility: existing running queries will not be affected by this change: they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`

Reviewer 4: ```suggestion
  Backward compatibility: existing running queries will not be affected by this change, and they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`
```

* Existing queries that reference a single GROUP BY column in the projection would fail if they were resubmitted, due to a duplicate column. The same existing queries will continue to run if already running, i.e. this is only a change for newly submitted queries. Existing queries will use the old query semantics.
* Any existing persistent queries, e.g. those created with `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT` or `INSERT INTO`, will be unaffected: their column names will not change. Pull queries will be unaffected. Push queries, which rely on auto-generated column names, may see a change in column names.

> * The ksqlDB server no longer ships with Jetty, which means Jetty-specific dependencies (e.g., certain login modules used for basic authentication) must be supplied via the KSQL_CLASSPATH environment variable when starting the server in order to be found.

Reviewer 4: ```suggestion
* The ksqlDB server no longer ships with Jetty. This means that when you start the server, you must supply Jetty-specific dependencies, like certain login modules used for basic authentication, by using the KSQL_CLASSPATH environment variable for them to be found.
```

* support more advanced suite of LIKE expressions ([#5013](https://github.com/confluentinc/ksql/pull/5013)) ([67cd9d9](https://github.com/confluentinc/ksql/commit/67cd9d91690039cc01f43438db1fbcd3fe4924ff))
* add COALESCE function ([#4829](https://github.com/confluentinc/ksql/pull/4829)) ([251c237](https://github.com/confluentinc/ksql/commit/251c237c455f4cf3843628f41e4bbf57795cfd12))
* add GROUP BY support for any key names ([#4899](https://github.com/confluentinc/ksql/pull/4899)) ([e7cbdfc](https://github.com/confluentinc/ksql/commit/e7cbdfcc8c9853e2ae6dfcaf670e04f07ccd5444)), closes [#4898](https://github.com/confluentinc/ksql/issues/4898)
* partition-by primitive key support (#4098) ([7addf88](https://github.com/confluentinc/ksql/commit/7addf8856a6d62a6890a5f2520eead26538233a6)), closes [#4098](https://github.com/confluentinc/ksql/issues/4098)
* add KsqlQueryStatus to decouple from KafkaStreams.State ([#5029](https://github.com/confluentinc/ksql/pull/5029)) ([e8cbcde](https://github.com/confluentinc/ksql/commit/e8cbcde548e1f3078b1396173a3f55f86ee20626))

> * add multi-join expression support ([#5081](https://github.com/confluentinc/ksql/pull/5081)) ([002cd5a](https://github.com/confluentinc/ksql/commit/002cd5ad6473cbb000291e32ebe5a459d2870b61))

Reviewer 5: Duplicated with the first item.

Author: Done

* Adds rate limiting to pull queries ([#4951](https://github.com/confluentinc/ksql/pull/4951)) ([6284111](https://github.com/confluentinc/ksql/commit/6284111652252b63cfc233bdfc3e08dbff983bbd))
* Do not allow access to new streaming endpoints using HTTP1.x ([#5193](https://github.com/confluentinc/ksql/pull/5193)) ([8b90035](https://github.com/confluentinc/ksql/commit/8b90035c6facf47d81dfd1616784b191141dce31))
* fail startup if command contains incompatible version ([#5104](https://github.com/confluentinc/ksql/pull/5104)) ([a1751b1](https://github.com/confluentinc/ksql/commit/a1751b1689532ef9abd436d5f27fe9aa8ff555ac))
* klip-14 - rowtime as pseduocolumn ([#5150](https://github.com/confluentinc/ksql/pull/5150)) ([d541420](https://github.com/confluentinc/ksql/commit/d541420acf61f0e0e7b25d91120886007dbdae01))

> * scatter gather query status from all servers in cluster for 'SHOW QUERIES [EXTENDED]' statement ([#4875](https://github.com/confluentinc/ksql/pull/4875)) ([7385a31](https://github.com/confluentinc/ksql/commit/7385a31dd33293b76e594649213da2abb81b8ddf))

Reviewer 5: We should spell out what this means for people who use it. Its written how it is implemented.

Author: Done. I used the title from the Issue instead of the PR.
```
'SHOW QUERIES [EXTENDED]' statement returns results for all nodes in the cluster 
```

    Backward compatibility: existing running queries will not be affected by this change, and they will continue to run with the same column names. Any statements executed after the upgrade will use the new names where no explicit alias is provided. Add explicit aliases to your statements if you require the old names, for example: `SELECT someStruct->someField AS SOMESTRUCT__SOMEFIELD, ...`

* Existing queries that reference a single GROUP BY column in the projection would fail if they were resubmitted, due to a duplicate column. The same existing queries will continue to run if already running, i.e. this is only a change for newly submitted queries. Existing queries will use the old query semantics.
* Any existing persistent queries, e.g. those created with `CREATE STREAM AS SELECT`, `CREATE TABLE AS SELECT` or `INSERT INTO`, will be unaffected: their column names will not change. Pull queries will be unaffected. Push queries, which rely on auto-generated column names, may see a change in column names.

> * The ksqlDB server no longer ships with Jetty. This means that when you start the server, you must supply Jetty-specific dependencies, like certain login modules used for basic authentication, by using the KSQL_CLASSPATH environment variable for them to be found.

Reviewer 5: How does this actually affect people? If it's important we should link to something more descriptive about how to fix it. If its a matter of just using plugins that are already being used today, then maybe it's not worth calling out.

Reviewer: Here's an example of how it affects users: https://github.com/confluentinc/ksql/pull/5307

Previously the Jetty LoginModule in the docs example would "just work," but now users have to provide the jar when starting the server.

Author: Done. I added a link to how this works.

Reviewer 5: Got it. Thanks!

* Improve error message for where/having type errors ([#5023](https://github.com/confluentinc/ksql/pull/5023)) ([23eb80d](https://github.com/confluentinc/ksql/commit/23eb80d06ea0275b28edac9b0fb5453ea4a966f8))
* improve handling of NULLs ([#5019](https://github.com/confluentinc/ksql/pull/5019)) ([c53dd68](https://github.com/confluentinc/ksql/commit/c53dd68bd912d79ced54403650ba13597ffe35a0)), closes [#4912](https://github.com/confluentinc/ksql/issues/4912)
* include lower-case identifiers among those that need quotes ([#3723](https://github.com/confluentinc/ksql/pull/3723)) ([#5139](https://github.com/confluentinc/ksql/issues/5139)) ([3bcbcf4](https://github.com/confluentinc/ksql/commit/3bcbcf45e1fc03245a57005d318133c568fb12c6))
* make endpoints available while waiting for precondition ([#5069](https://github.com/confluentinc/ksql/pull/5069)) ([1136162](https://github.com/confluentinc/ksql/commit/113616260cc0db89124330ed5e935e4b9e5af281))
* Make sure internal client is configured for TLS ([#5059](https://github.com/confluentinc/ksql/pull/5059)) ([37c7713](https://github.com/confluentinc/ksql/commit/37c7713920c7654c9b5c2f81ed72225233e1176c))

> * NPE in latest_by_offset if first two elements processed are both null… ([#4975](https://github.com/confluentinc/ksql/pull/4975)) ([4c72f93](https://github.com/confluentinc/ksql/commit/4c72f93813af19f6fd074b3a691018434731d545))

Reviewer: nit: remove the trailing ellipsis.

Author: Done
